esp-hal-common/src/interrupt/riscv.rs

```diff
@@ -417,6 +435,10 @@ pub unsafe extern "C" fn start_trap_rust_hal(trap_frame: *mut TrapFrame) {
         let interrupt_priority = _handle_priority();
 
         let code = mcause::read().code();
+
+        #[cfg(clic)]
+        let code = (code & 0b1111_1111_1111) - 16;
```

Comment:
Could we get a comment explaining what this is doing?

Reply:
Also maybe its worth trying to use `EXTERNAL_INTERRUPT_OFFSET` (If that's what we're doing here) as that would also help explain whats going.

Reply:
Good point! Thanks